library/src/CMakeLists.txt

```diff
@@ -110,23 +110,20 @@ if( NOT USE_CUDA )
     endif( )
   endif( )
 
-  target_link_libraries( hipsolver PRIVATE roc::rocblas roc::rocsolver hip::host )
+  target_link_libraries( hipsolver
+    PUBLIC
+      hip::host
+    PRIVATE
+      roc::rocblas
+      roc::rocsolver
+  )
 
   if( CUSTOM_TARGET )
     target_link_libraries( hipsolver PRIVATE hip::${CUSTOM_TARGET} )
   endif( )
 
 else( )
-  target_compile_definitions( hipsolver PRIVATE __HIP_PLATFORM_NVCC__ )
-
-  target_link_libraries( hipsolver PRIVATE ${CUDA_cusolver_LIBRARY} )
-
-  # External header includes included as system files
-  target_include_directories( hipsolver
-    SYSTEM PRIVATE
-      $<BUILD_INTERFACE:${HIP_INCLUDE_DIRS}>
-      $<BUILD_INTERFACE:${CUDA_INCLUDE_DIRS}>
-  )
+  target_link_libraries( hipsolver PRIVATE hip::host ${CUDA_cusolver_LIBRARY} )
```

Comment:
`hip::host` should be `PUBLIC` but HIP needs to export the target. We need SWDEV-243403 to be able to do that.